### PR TITLE
WIP: DES-2549: Extend journal look-back by 2 weeks

### PR DESCRIPTION
**Pull request by MarkPatt89**
## Description and relevant Jira numbers
- Allow date traversal in the journal of up to 2 weeks.
- Extend caching period.

Screenshot 2019-06-24 at 09 15 43

Screenshot 2019-06-24 at 09 15 33

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [x] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [ ] PR link added to JIRA
- [x] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/pages/journal/components/journal-navigation/__tests__/journal-navigation.spec.ts [hunk 1]
```diff
@@ -65,10 +65,10 @@ describe('JournalNavigationComponent', () => {
     });
 
     describe('selected date is today', () => {
-      it('does not show previous day button', () => {
+      it('does show previous day button', () => {
         fixture.detectChanges();
         const previousDayContainer: DebugElement = componentEl.query(By.css('#previous-day-container'));
-        expect(previousDayContainer).toBeNull();
+        expect(previousDayContainer).toBeDefined();
```
Comment (johnmorleybjss) [inline src/pages/journal/components/journal-navigation/__tests__/journal-navigation.spec.ts]: Careful of the distinction between `toBeNull()` and `toBeDefined()` when querying the component fixture. The absence of the queried selector in the DOM yields `null` which passes a check of `toBeDefined` - always use null when testing a query result.

File: src/pages/journal/components/journal-navigation/__tests__/journal-navigation.spec.ts [hunk 2]
```diff
@@ -149,6 +149,7 @@ describe('JournalNavigationComponent', () => {
 
       it('does not shows next day button', () => {
         fixture.detectChanges();
+        console.log(`selectedDay is ${selectedDay}`);
```
Comment (johnmorleybjss) [inline src/pages/journal/components/journal-navigation/__tests__/journal-navigation.spec.ts]: Extra log.

File: src/pages/journal/components/journal-navigation/__tests__/journal-navigation.spec.ts (hunk 1 above)
Comment (MarkPatt89) [reply src/pages/journal/components/journal-navigation/__tests__/journal-navigation.spec.ts]: I'll be aware of this in the future. Thanks for the explanation.

Comment (MarkPatt89) [reply src/pages/journal/components/journal-navigation/__tests__/journal-navigation.spec.ts]: Refactored with `.not.toBeNull()`

**Comment by ammarhaiderbjss** [comment]
Is there another task to amend the `daysToCacheJournalData` config var?

**Comment by MarkPatt89** [comment]
> Is there another task to amend the `daysToCacheJournalData` config var?

I'd missed this one Ammar, thanks. I'll amend it now.

**Comment by ammarhaiderbjss** [comment]
> > Is there another task to amend the `daysToCacheJournalData` config var?
> 
> I'd missed this one Ammar, thanks. I'll amend it now.

Cool, and will the config service be updated as well?

**Pull request merged**